esp-wifi/src/ble/btdm.rs

```diff
@@ -224,19 +223,15 @@ unsafe extern "C" fn queue_recv(queue: *const (), item: *const (), block_time_ms
     let item = item as *mut _;
 
     loop {
-        let res = critical_section::with(|_| {
-            memory_fence();
-
-            critical_section::with(|_cs| {
-                let queue = queue as *mut RawQueue;
-                if (*queue).try_dequeue(item) {
-                    trace!("received from queue");
-                    1
-                } else {
-                    0
-                }
-            })
-        });
+        let res = {
+            let queue = queue as *mut RawQueue;
```

Comment:
This pattern made sense with the critical section but now we should be able to simplify it I hope.